### DISCUSSION TITLE
Troubleshooting XAML Errors

### DISCUSSION THREAD

**Discussion by LadderLogic** [microsoft/microsoft-ui-xaml#7368] [Q&A]
We are looking for a way to troubleshoot XAML Exceptions that happen at runtime, beyond just the binding failures. At the moment a fatal COM exception is thrown with very little information such as "Invalid value to _unknown_ to Property _Opacity_" but there is no way to tell where this error is coming from or which object this belongs to. 

Is there any tooling or trick to improve the verbosity of the exceptions for troubleshooting purposes?

**Comment by krschau** [comment]
Unhelpful runtime exception errors is definitely a pain point we're aware of, and something we've started working on. @Scottj1s, does this look like something that falls under what you're doing, or is it a different category?

(If so, internally tracked as https://task.ms/39696500)
